libs/ngxtension/svg-sprite/src/svg-sprite.ts

```diff
@@ -0,0 +1,471 @@
+ *
+ * ```ts
+ * provideSvgSprites(
+ * createSvgSprite({
+ * name: 'fa-brands',
+ * baseUrl: 'assets/fontawesome/sprites/brands.svg',
+ * }),
+ * );
+ * ```
+ *
+ * The `name` property can reference any arbitrary value, but should be unique, since you can register multiple different svg sprites.
+ *
+ * The `sprite` input of the `NgxSvgSpriteFragment` should reference the `name` property of a provided sprite.
+ *
+ * ### Auto View Box
+ *
+ * When a symbol of an svg sprite is rendered the `viewBox` attribute or `height` and `width` _should_ be set. The `svg` element does not copy/use the `viewBox` attribute of the symbol in the svg sprite, therefore the svg will have default dimensions of 300x150 px, which is probably not preferred.
+ *
+ * Per default when an svg sprite is registered, the svg sprite is fetched with js in addition. `NgxSvgSpriteFragment` will copy the `viewBox` attribute of the symbol to its host.
+ *
+ * This behavior can be disabled.
+ *
+ * #### Disable via DI
+ *
+ * Auto View Box is disabled for the svg sprite.
+ *
+ * ```ts
+ * provideSvgSprites(
+ * createSvgSprite({
+ * name: 'fa-brands',
+ * baseUrl: 'assets/fontawesome/sprites/brands.svg',
+ * autoViewBox: false,
+ * }),
+ * );
+ * ```
+ *
+ * #### Disable via `autoViewBoxDisabled` Input
+ *
+ * Auto View Box is disabled for a `svg` element, when the `autoViewBoxDisabled` input is set to `false`.
+ *
+ * ```html
+ * <svg fragment="github" sprite="fa-brands" autoViewBoxDisabled></svg>
+ * ```
+ *
+ * #### Disable via `viewBox` Attribute
+ *
+ * Auto View Box is disabled for a `svg` element, when the `viewBox` attribute already is defined.
+ *
+ * ```html
+ * <svg fragment="github" sprite="fa-brands" viewBox="0 0 32 32"></svg>
+ * ```
+ *
+ * ### Classes
+ *
+ * When the `classes` function is set, a list of classes will be added by the `NgxSvgSpriteFragment` to its host.
+ *
+ * ```ts
+ * provideSvgSprites(
+ * createSvgSprite({
+ * name: 'my-sprite',
+ * baseUrl: 'path/to/my/sprite.svg',
+ * classes: (fragment) => ['some-class', `some-other-class-${fragment}`],
+ * }),
+ * );
+ * ```
+ *
+ * ### Url
+ *
+ * Per default when using the `createSvgSprite` function, the `url` will return `'${baseUrl}#${fragment}'`. This can be overwritten:
+ *
+ * ```ts
+ * provideSvgSprites(
+ * createSvgSprite({
+ * name: 'my-sprite',
+ * baseUrl: 'path/to/my/sprite.svg',
+ * url: (baseUrl, fragment) => `${baseUrl}#some-prefix-${fragment}`,
+ * }),
+ * );
+ * ```
+ */
+@Directive({
+	selector: 'svg[fragment]',
+	standalone: true,
+})
+export class NgxSvgSpriteFragment implements OnInit {
+	/**
+	 * @ignore
+	 */
+	private readonly element = inject(ElementRef)
+		.nativeElement as SVGGraphicsElement;
+
+	/**
+	 * @ignore
+	 */
+	private readonly injector = inject(Injector);
+
+	/**
+	 * @ignore
+	 */
+	private readonly sprites = inject(NgxSvgSprites);
+
+	/**
+	 * @ignore
+	 */
+	public ngOnInit() {
+		runInInjectionContext(this.injector, () => {
```

Comment:
comment (non-blocking): there's `autoEffect` in `ngxtension/auto-effect` that you can use if you don't want to inject the `Injector` and `runInInjectionContext` manually 😛

Reply:
Awesome!